src/intfuncs.c

```diff
@@ -620,6 +620,293 @@ Obj FuncINT_STRING ( Obj self, Obj string )
 
 /****************************************************************************
 **
+*F SmallInt Bitmask operations
+*
+*  Since all our functions have special handlers and no BODY, NLOC is unused,
+*  so we repurpose it.
+*
+*/
+
+#ifdef SYS_IS_64_BIT
+#define LENSTART_WIDTH 8
+#define WIDTH_WIDTH 6
+#define MAX_FIELDS 10
+#else
+#define LENSTART_WIDTH 8
+#define WIDTH_WIDTH 5
+#define MAX_FIELDS 6
+#endif
+
+static inline UInt get_start(UInt mask) {
+  return mask & ((1L << LENSTART_WIDTH) -1);
+}
+
+static inline UInt get_len(UInt mask) {
+  return mask >> LENSTART_WIDTH;
+}
+
+static inline UInt make_mask(UInt start, UInt len) {
+  return len << LENSTART_WIDTH | start;
+}
+
+static inline UInt get_width(UInt mask, UInt i) {
+  return (mask >> (WIDTH_WIDTH*(i-1)) & ((1 << WIDTH_WIDTH) -1));
+}
+
+static inline UInt set_width(UInt mask, UInt i, UInt width) {
+  return (((mask >> (WIDTH_WIDTH*i) << WIDTH_WIDTH) | width) << (WIDTH_WIDTH*(i-1))) | (mask & ((1L << (WIDTH_WIDTH*(i-1))) -1L));
+}
+
+static Obj doMaskGetter(Obj self, Obj data) {
+  UInt mask = NLOC_FUNC(self);
+   UInt start = get_start(mask);
+       UInt len = get_len(mask); 
+  if (!IS_INTOBJ(data))
+    ErrorMayQuit("Mask getter: argument must be small integer",0,0);
+  UInt x = INT_INTOBJ(data);
+  /* return INTOBJ_INT(__builtin_ia32_bextr_u64(x,mask));*/
+    return INTOBJ_INT((x >> start) & ((1 << len) -1)); 
+}
+
+static Obj doMaskSetter(Obj self, Obj data, Obj val) {  
+  UInt mask = NLOC_FUNC(self);
+  UInt start = get_start(mask);
+  UInt len = get_len(mask);
+  if (!ARE_INTOBJS(data,val))
+    ErrorMayQuit("Mask Setter: both arguments must be small integers",0,0);
+  UInt x = INT_INTOBJ(data);
+  UInt y = INT_INTOBJ(val);
+  GAP_ASSERT(y < (1 << len));
+  return INTOBJ_INT((((x >> (start+len)) << len | y) << start) |
+		    (x & ((1<<start) -1)));
+}
+
+/*
+* For the builders we also repurpose BODY_FUNC to hold a GAP integer telling
+* us how many fields there are
```

Comment:
"also"? As in: something else was "also" repurposed? But what? 

It would be really nice if there was a documentation comment before this code which gave at least a brief overview of what is stored where and why...